# Chapter 02 User Input, Type Casting & String Data Type 

### Topics
- user input & multi user input
- Type casting
- Literals
- String concatenation
- String Formatting
    - String formatting  using format function
    - String formatting using f 
    - string multiple lines formatting using tripple quoates
- Accessing Substring from a String
    - String indexing
    - string Slicing
- String Edit & Delete
- string Operations
- String methods

## User Input
- In programming, we might want to take the input from the user. In Python, we can use the input() function.
- input function take input from user on console. 
- the input given is called console input/ runtime input

In [1]:
input()

155


'155'

- when we simply use input() function there is a problem for the user. Because they couldn't understand what to enter in the input box like name, age, etc.
- for Solution we add a prompt which is exactly seen to the user.

In [2]:
input("Apna Naam Batao: ")

Apna Naam Batao: Muhammad Huzaifa


'Muhammad Huzaifa'

#### Syntax of Input Function
- input(prompt)
- Here, prompt is the string we wish to display on the screen for User. It is optional.

In [3]:
name = input("Tell Me Your Good Name: ")
print("Your Good Name is: ", name)

Tell Me Your Good Name: Muhmmad Huzaifa
Your Good Name is:  Muhmmad Huzaifa


- **The input given from console will always be a string either you input a
string value or integer value**

In [4]:
name = input("Tell me your good name sir! ")

Tell me your good name sir! Nasir Hussain


In [5]:
print(name)
print(type(name))

Nasir Hussain
<class 'str'>


In [5]:
age = input("Tell me your age>")
print(age)
print(type(age))
# still the type of age is string even we enter a integer value in input

Tell me your age>22
22
<class 'str'>


- Write a program that take two number as input and find addition of the two numbers.

In [7]:
first_num = input("Enter the First Number: ")
second_num = input("Enter the Second Number: ")
result = first_num + second_num
print(result)

Enter the First Number: 12
Enter the Second Number: 15
1215


In [9]:
print(type(first_num))
print(type(second_num))

<class 'str'>
<class 'str'>


- we can solve this mathematical addition issue with type casting in later

### Can we give more than one input from console?

- No we can not give more than one input from console. 
- But we can use different techniquues to break single input into more than one input 

In [13]:
asad,usman  = input("Tell me your age: ").split(',')

Tell me your age: 23,36


- 23,36 is inputted in a single string like '23,36'. 
- But we use split() function to split this single string on "," to break it into two string.
- since split will break "23,36" into  '23','36' and assigned it into two variables
- asad <<<== 23 & usman <<<=== 36

- Note: since our string value will be borken down into 2 values we will need two variables to catch the values. 

In [14]:
print(asad)
print(usman)

23
36


## Type Casting
- changing the data type of a value to another data type is called type casting
    - implicit type casting
    - explicit type casting

### Implicit Type Casting
- In some situations, Python automatically converts one data type to another. This is known as implicit type casting.

In [16]:
# implicit type casting Example
6 + 5.5
# here python automatically convert int data type to float data type.

11.5

In [18]:
# implicit type casting example
6 + 7+6j
# here python automatically convert int data type to complex data type.

(13+6j)

### Explicit Type Casting
- In explicit type casting, users convert the data type of an object to the required data type.
- We use built-in functions like int(), float(), str(), etc. to perform explicit type casting.

In [20]:
# ----> int() <----
# float to int
print(int(46.5))
# string to int
print(int('65'))

46
65


In [21]:
# string to int
int('Karachi') 

ValueError: invalid literal for int() with base 10: 'Karachi'

In [26]:
# ----> float() <----
# int to float
print(float(4564))
# string to float
print(float('45.65'))

4564.0
45.65


In [24]:
# string to float
float('Lahore')

ValueError: could not convert string to float: 'Lahore'

In [29]:
# ----> str() <----
# int to string
str(45)
print(type(str(45)))


<class 'str'>


In [30]:
str('4.5')

'4.5'

- type casting is not perminent Like

In [32]:
a = 45.5
int(a)

45

In [33]:
print(a)

45.5


In [35]:
# Now it is perminent
a = int(45.5)
print(a)

45


- now we solve the previous addition program with explicit type casting

In [31]:
first_num = int(input("Enter the First Number: "))
second_num = int(input("Enter the Second Number: "))
result = first_num + second_num
print(result)

Enter the First Number: 45
Enter the Second Number: 65
110


## Literals in python
- literals is a raw data or value given to a variable.
- In python, there are various typs of literals
    1. Numeric Literals
    2. String Literals
    3. Boolean Literals
    4. Special Literals

### 1. Numeric Literals

- **int Literals**
    - Decimal ------>  5, 10, -68 -----> Regular numbers.
    - Binary ------->  0b101, 0b11 ----> Start with 0b.
    - Octal -------->  0o13 -----------> Start with 0o.
    - Hexadecimal ---> 0x13 -----------> Start with 0x.

In [4]:
a = 10 # Decimal
b = 0b1001 # binary
c = 0o13 # octal
d = 0x12A # hexadecimal

# Float Literals
x = 10.56 # Normal float
y = 1.6e3 # hightest exponential value
z = 1.6e-3 # lowest exponential value

print(a, b, c, d)
print(x, y, z)

10 9 11 298
10.56 1600.0 0.0016


### 2. String Literal

In [9]:
string = 'This is Single Quote String.'
strings = "This is double Quote String."
char = "H" # single character
multi_line = """This is multiline String used to get large values."""
unicode = u"\U0001f600\U0001F606\U0001F923" # use to add emogis
raw_str = r"raw \n \t <br> string" # use for add html code

print(string, strings, char, multi_line)
print(unicode)
print(raw_str)

This is Single Quote String. This is double Quote String. H This is multiline String used to get large values.
😀😆🤣
raw \n \t <br> string


### 3. Boolean Literals

In [3]:
a = True + 5 # the value of True by default 1 in python
#    1 + 5 = 6
b = False + 16 # the value of Flase by default 0 in python
#    0 + 16 = 16
print(a)
print(b)

6
16


### 4. Special Literals

- None is a special literal which is used when we could know the value of a variable but we want to use the variable in later of the program we assign it None. None is nothing just a sign of absent.

In [4]:
h = None
print(h)

None


## String Concatination
- it is a method of joining the two string togeather.
- the (+) operator is used for string concatination.

In [7]:
fName = "Muhammad"
lName = "Huzaifa"
fName + lName

'MuhammadHuzaifa'

In [8]:
"Pakistan" + "Zindabad"

'PakistanZindabad'

In [9]:
"Mohsin" + 23

TypeError: can only concatenate str (not "int") to str

In [10]:
# to solve this we can typecast the second integer value into string with str()
"Mohsin" + str(23)

'Mohsin23'

In [12]:
name = input("Enter your name: ")
YearOfBirth = input("Enter Your Year of Birth: ")
# Here year of birth although an integer value
# but due to console input it will be assign as string value
print("Your name is " + name + " and your Year of Birth is " + YearOfBirth) 

Enter your name: Muhammad Usama
Enter Your Year of Birth: 1999
Your name is Muhammad Usama and your Year of Birth is 1999


In [18]:
student_name = input("Enter your name: ")
student_age = input("Enter your age: ")

Enter your name: Muhammad Tahir
Enter your age: 23


In [19]:
after_10_years = 10
student_age + after_10_years
# str           int

TypeError: can only concatenate str (not "int") to str

In [15]:
student_name = input("Enter your name: ")
student_age = int(input("Enter your age: ")) 
# common sense says age should be integer value so we can typecast to int in input function

Enter your name: Muhammad Tahir
Enter your age: 23


In [17]:
after_10_years = 10
student_age + after_10_years
# int           int

33

- when we take age as an input and the user add age in floating number so there an error generate

In [20]:
student_age = int(input("Enter your age: ")) 

Enter your age: 23.5


ValueError: invalid literal for int() with base 10: '23.5'

- when a user enter age in float like 23.5
- the input convert it into string like '23.5'
- and then int() function typecast and remove string quotation like 23.5 
- but we saw that the value is float and we have typecast it as an int so there is an error generate
- to solve we can typecast it as a float like

In [21]:
student_age = float(input("Enter your age: ")) 

Enter your age: 23.5


- but we don't need float number we need an integer value so we can again typecast it int() after float()

In [22]:
student_age = int(float((input("Enter your age: "))))

Enter your age: 23.5


In [23]:
print(student_age)

23


In [24]:
student_name = input("Enter your name: ")
student_age = int(input("Enter your age: "))

print("Your name is " + student_name + " and your age is " + student_age) 

Enter your name: Muhmmad Tahir
Enter your age: 23


TypeError: can only concatenate str (not "int") to str

In [25]:
student_name = input("Enter your name: ")
student_age = int(input("Enter your age: "))

print("Your name is " + student_name + " and your age is " + str(student_age)) 
# solve error by typecast student age into string with str()

Enter your name: Muhammad Tahir
Enter your age: 23
Your name is Muhammad Tahir and your age is 23


## String Formatting
- Concatenation joins two strings where as formatting is a method to format a string.
- We don't need to type cast values in formatting which we used in concatination.
- Formatting is more convenient and easy to handle the values
- There are two methods of string formatting.
    - format()
    - f-method

### String Formatting using format()

- {} braces are called palace holders.
- value used in {} place holders will be supplied from format() function in sequence
- 1st place holder can hold 1st value of format() function 2nd place holder can hold 2nd value of format() function respectively. 
- format() function can have hard coded value or a variable

In [8]:
# variable values
user_name = input("Enter your name ")
user_age  = int(input("Enter your age ")) 

print("User name is {} and age is {}".format(user_name,user_age))

Enter your name Muhammad Hamza
Enter your age 12
User name is Muhammad Hamza and age is 12


In [9]:
# hard coded values
print("User name is {} and age is {}".format("Muhmmad Hamza",12))

User name is Muhmmad Hamza and age is 12


In [12]:
print("User name is {} and age is {}".format("Muhammad Fayyaz", 45))
                                        #           0            1

User name is Muhammad Fayyaz and age is 45


In [16]:
# we can also add index number in place holders
print("User name is {0} and age is {1}".format("Muhammad Fayyaz", 45))

User name is Muhammad Fayyaz and age is 45


- if we have more place holders than values supplied in format() function it will throw an error. 

In [15]:
print("Student name is {} and student age is {}. Student enroll in {} department".format("Tahir", 23))

IndexError: Replacement index 2 out of range for positional args tuple

- Using index of values in palce holder allows to reuse of same values in different placeholders in a string

In [17]:
print("my name is {0} and i am {1} years old. {0} is my name and age is {1}".format("Huzaifa", 22))

my name is Huzaifa and i am 22 years old. Huzaifa is my name and age is 22


### String Formatting using f-Method

- In f methods we use a f before starting a string. 
- {}{} place holders can directly take the variables. 
- This allows more flexibility in formatting a string. 

In [22]:
# using varibles in place holders
user_name = input("Enter your name ")
birth_date = input("Enter your birth date ")

print(f"Your name is {user_name} and your birth date is {birth_date}")

Enter your name Muhammad Huzaifa
Enter your birth date 19-08-2001
Your name is Muhammad Huzaifa and your birth date is 19-08-2001


In [2]:
# using hard coded values in place holder
print(f"Hello World my name is Muhmmad Huzaifa. I was born in year {2001} in karachi. my age is {22}. my matric percentage is {77.5}%.")

Hello World my name is Muhmmad Huzaifa. I was born in year 2001 in karachi. my age is 22. my matric percentage is 77.5%.


### Multiple line strings 
- A tripple quote string allows a multiple line string with spaces and line changes as it is as you supply in string

In [4]:
name = input("Enter Your Name: ")
f_name = input("Enter Your Father Name: ")
course = input("Enter Your Course Name: ")
batch_Year = int(input("Enter Your Batch Year"))
centre = input("Enter Your Centre: ")

Enter Your Name: Muhammad Huzaifa
Enter Your Father Name: Muhammad Akram
Enter Your Course Name: Artificial Intelligence (AI)
Enter Your Batch Year2022
Enter Your Centre: Sir Sayed University of Eng. & Technology


In [55]:
print(f"""
 _________________________________________________
|                                                 | 
|                      ID Card                    |
| ________________________________________________|
|                                                 | 
|                       PIAIC                     |
|                                                 |
| Name : {name          }                         |
| Father Name: {f_name      }                     |
|                                                 |
| Course: {course                    }            |       
| Batch Year: {batch_Year                       }                                |
| Centre: {centre}| 
|                                                 |
|_________________________________________________|

""")


 _________________________________________________
|                                                 | 
|                      ID Card                    |
| ________________________________________________|
|                                                 | 
|                       PIAIC                     |
|                                                 |
| Name : Muhammad Huzaifa                         |
| Father Name: Muhammad Akram                     |
|                                                 |
| Course: Artificial Intelligence (AI)            |       
| Batch Year: 2022                                |
| Centre: Sir Sayed University of Eng. & Technology| 
|                                                 |
|_________________________________________________|




## Accessing Substring from a String
- accessing means to extract and substring means small part of a string.
- it means we can extract a small part of string from a complete string.
- this can be done with two methods
    - String indexing
    - String Slicing

### String Indexing
- In computer programming, a string is a sequence of characters. For example, "hello" is a string containing a sequence of characters 'h', 'e', 'l', 'l', and 'o'
- every character has an index position which is starting from 0.
- It means first letter in string has 0 index number which is 'h'.
- we can use [] brackets to access substring according to its index value.

In [4]:
message = "Hello World!"
#          012345678901    index 0 to 11
print(message)

Hello World!


In [5]:
message[0]

'H'

In [6]:
message[1]

'e'

In [7]:
message[6]

'W'

In [10]:
message[12]

IndexError: string index out of range

### Types of indexing
- There are two types of indexing in python
    - positive indexing
    - negative indexing

#### Positive indexing
- it start from 0 to onwards like "01234...."
- it moves from left to right

In [11]:
# positive indexing
institute = "Dawood University"
#            01234567890123456  index from 0 to 16
print(institute[5])
print(institute[14])

d
i


#### Negative indexing
- it start from -1 to onwards like '-1-2-3-4-5...'
- it moves form right to left
- it mostly used to find last character of a string.

In [12]:
# Negative indexing
# -2-1-0-9-8-7-6-5-4-3-2-1-0-9-8-7-6-5-4-3-2-1
course = "Artificial Intellignce"
#   index from -1 to -22
course[-1]

'e'

In [13]:
course[-22]

'A'

In [14]:
course[-10]

'n'

In [15]:
course[-23]

IndexError: string index out of range

### String Slicing
- Access a range of characters in a string by using the slicing operator colon **( : )**.
- syntax of slicing --> string_variable[starting_point : ending_point-1] 

In [2]:
string = "Pakistan is a beautiful country."
print(string)

Pakistan is a beautiful country.


In [3]:
# slicing with (:)
string[0:7] 
# we always use n+1 in ending point because it not select the last index number which is specified
# like 'pakistan' here pakistan has 0 to 7 index but when we slice it will give string upto 6 index
#       01234567 

'Pakista'

In [4]:
# slicing with (:) for getting 'pakistan' substring
print(string[0:8])

Pakistan


In [5]:
# slicing with (:) for getting 'beautiful' substring
print(string[14:23])

beautiful


In [9]:
# Variations of slicing
# starting point is given but ending point is not considered so it will slice till to end.
print(string[9:])

is a beautiful country.


In [10]:
# Variations of slicing
# starting point is not considered but ending point given it will slice from starting to index point.
print(string[:8])

Pakistan


In [12]:
# Variations of slicing
# starting point and ending point is not considered so it will slice from starting to till the end.
print(string[:])

Pakistan is a beautiful country.


- we have one more thing in slicing which steps in slicing like this string[starting:ending:steps]
- it is used to skip the steps when slicing from staring to the end

In [13]:
string[0:10:2]

'Pksa '

In [14]:
string[::2]

'Pksa sabatflcuty'

- when we use positive indexing we can't use negative steps. it will return empty string.

In [16]:
string[0:8:-1]

''

- but we use negative indexing and positive steps like this below

In [20]:
string[-14:-1:2]

'tflcuty'

- A famous syntax in python to reverse a whole string in one line of code.

In [21]:
string[::-1]

'.yrtnuoc lufituaeb a si natsikaP'

In [22]:
message = "Hello World"
message[::-1]

'dlroW olleH'

## String Edition & Deletion

In [23]:
msg = "Pakistan Zindabad"
print(msg)

Pakistan Zindabad


- if i want to change first letter of string from 'P' to 'B' can it works?

In [24]:
msg[0] = "B"

TypeError: 'str' object does not support item assignment

- No string does not support editing to change a particular character
- beacause string is a immutable data type.
- immutable means no change

- for this purpose to change a character in a string we can reassign the complete string.

In [25]:
msg = "pakistan zindabad!"
print(msg)

pakistan zindabad!


- also you can not add a character to existing string

In [26]:
print(msg)
# pakistan zindabad!
# 123456789012345678 index 0 to 18 but want add in index 19.

pakistan zindabad!


In [27]:
msg[19] = "love Pakistan"

TypeError: 'str' object does not support item assignment

#### Deletion in String
- for deletion we can use del keyword to delete the complete string.

In [6]:
note = "Please Delete this Message!."
print(note)

Please Delete this Message!.


In [7]:
# deletion in string 
del note

In [8]:
print(note)

NameError: name 'note' is not defined

In [9]:
# similar to above we can not delete one character from a string.
note = "Don't Delete Please!"
del note[0]

TypeError: 'str' object doesn't support item deletion

In [10]:
del note[0:3]

TypeError: 'str' object does not support item deletion

## String Operation
- Artithmetic Operations
- Comparision operations
- Membership operators

In [11]:
# Airthmetic Operation have two operators used in string (+) and (*)
# (+) which know n for string cancatination
"Hello" + " " + "World!"

'Hello World!'

In [14]:
# (*) operator is used in String
print("*" * 30)

******************************


In [15]:
print("Hello World "*3)

Hello World Hello World Hello World 


In [16]:
# Comparision operators are used to compare two string with each other and give boolean result
"Hello" == "Hello"

True

In [17]:
"Hello" == "World"

False

In [18]:
"Hello" != "Germany"

True

In [19]:
"hello" != "hello"

False

In [20]:
"Germany" > "New Zeland"
# Lexiographicaly which means in dictionary which word comes first
# in python which word comes first is smaller than the word comes last in dictionary

False

In [21]:
"New Zeland" < "England"

False

In [22]:
"USA" > "Pakistan"

True

In [23]:
# Membership operators used to check substring is present in a string
name ="Muhammad Huzaifa"
"Huzaifa" in name

True

In [24]:
"M" in name

True

In [25]:
"g" in name

False

In [26]:
"Muhammad" not in name

False

In [28]:
"F" not in name

True

In [29]:
"zafi" not in name

True

## String Methods/Functions

### Common Function
- len()
- max()
- min()
- sorted()

In [1]:
# len() ---> used to find length of characters in a string
city = "karachi"
len(city)

7

In [2]:
# max() ---> gives you biggest character according to ASCII Values
max(city)

'r'

In [3]:
# min() ---> gives you smallest character according to ASCII Values
min(city)

'a'

In [4]:
# sorted() ---> it will sort string character in ascending order according to ASCII values
# it always return a list in output
sorted(city)

['a', 'a', 'c', 'h', 'i', 'k', 'r']

In [5]:
# For descending order 
sorted(city, reverse = True)

['r', 'k', 'i', 'h', 'c', 'a', 'a']

### Capitalize/Title/Upper/Lower/Swapcase

In [10]:
# capitalize() ---> it is uesd to capital first alphabet or character in a string
city.capitalize()

'Karachi'

In [11]:
# but it will not change original string
print(city)

karachi


In [12]:
"today is sunday".capitalize()

'Today is sunday'

In [14]:
# Title() ---> it will capaital first letter of each word in the string
"today is sunday".title()

'Today Is Sunday'

In [15]:
# Upper() ---> it will converts whole string to uppercase
city.upper()

'KARACHI'

In [16]:
# lower() ---> it will converts whole string to lowercase
"PIAIC".lower()

'piaic'

In [17]:
# swapcase() ---> it will converts lowercase to uppercase & Uppercase to lowercase in a string
"KaRaChi".swapcase()

'kArAcHI'

### Count

In [18]:
# count() ---> it will gives you count/occurance of a substring in a string
"Today is Sunday".count("S")

1

In [20]:
# it can also use for multiple characters
"Today is Raining".count("Rain")

1

In [21]:
"Today is Raining".count("rain")

0

In [22]:
"Today is Raining".count("ing")

1

In [23]:
"Today is Raining".count("i")

3

### Find/Index

In [24]:
# find() ---> it will gives you the index of first occurrence of substring
"Today is Sunday".find("S")

9

In [25]:
"Today is Sunday".find("Sunday")

9

In [26]:
"Today is Sunday".find("day")

2

In [27]:
"Today is Sunday".find("a")

3

In [29]:
# if you find any substring that is not existing it will gives you -1
"Today is Sunday".find("b")

-1

In [30]:
# index() ---> similar to find() use to find index of a substring
"Today is Sunday".index("S")

9

In [31]:
# differnce between find() and index() is that 
# if a substring is not exist in string find() will return -1
# but if a substring is not exist in string index() it will return an error
"Today is Sunday".index("b")

ValueError: substring not found

### endswith/startwith

In [32]:
# endswith() ---> Checks if String Ends with the Specified substring
"Today is Raining".endswith("g")

True

In [33]:
"Today is Raining".endswith("ing")

True

In [34]:
"Today is Raining".endswith("f")

False

In [35]:
# startswith() ---> Checks if String Starts with the Specified substring
"Today is Raining".startswith("Today")

True

In [1]:
# startswith() ---> Checks if String Starts with the Specified substring
"Today is Raining".startswith("T")

True

### isalmum/isalpha/isdecimal/isdigit/isidentifier 

In [3]:
# isalnum() ---> it checks Checks Alphanumeric Character in string
"House 298".isalnum()

True

In [4]:
# if string contains special charcaters it will return false
"House # 298".isalnum()

False

In [6]:
# isalpha() ---> Checks if All Characters are Alphabets in the string
"IamlivinginHouse".isalpha()

True

In [7]:
# it gives false we use space between words
"I am living in House".isalpha()

False

In [8]:
# isdecimal() --> Checks Decimal Characters in string
"205".isdecimal()

True

In [10]:
"205A".isdecimal()

False

In [12]:
# isdigit() ---> Checks Digit Characters in string
"2023".isdigit()

True

In [14]:
"2A".isdigit()

False

In [15]:
# isidentifier() ---> Checks for Valid Identifier
"first name".isidentifier()

False

In [16]:
# we can not use space in variable name 
"first_name".isidentifier()

True

### Split

In [17]:
# split() ---> it will Splits String from Left
# it will converts all words into a list.
"My Favorite Cricketer is Shahid Afridi".split()

['My', 'Favorite', 'Cricketer', 'is', 'Shahid', 'Afridi']

In [19]:
# we can split with any character which is exist in string
"My Favorite Cricketer is Shahid Afridi".split("i")
# when interperator read i in string it split and move next

['My Favor', 'te Cr', 'cketer ', 's Shah', 'd Afr', 'd', '']

In [21]:
# we can split with any character which is not exist in string. it will return whole string
"My Favorite Cricketer is Shahid Afridi".split("g")

['My Favorite Cricketer is Shahid Afridi']

### join

In [26]:
# join() ---> Returns a Concatenated String
text = ['Python', 'is', 'a', 'fun', 'programming', 'language']
# join elements of text with space
print(' '.join(text))

Python is a fun programming language


In [27]:
text = ['Python', 'is', 'a', 'fun', 'programming', 'language']
# join elements of text with space
print('/'.join(text))

Python/is/a/fun/programming/language


In [28]:
text = ['Java', 'is', 'not', 'a', 'fun', 'programming', 'language']
# join elements of text with space
print('-'.join(text))

Java-is-not-a-fun-programming-language


### Replace

In [29]:
# replace() ---> it replaces each matching occurrence of a substring with another string.
"My name is Muhammad Huzaifa".replace("Huzaifa", "Tahir")

'My name is Muhammad Tahir'

In [30]:
# it will not change in the original value of variable until reassign
text = "My favorite Singer is Atif Aslam."
text.replace("Atif Aslam", "Arijit Singh")

'My favorite Singer is Arijit Singh.'

In [31]:
# now i print text
print(text)

My favorite Singer is Atif Aslam.


In [32]:
# now i will reassign the value to change
text = "My favorite Singer is Atif Aslam."
text = text.replace("Atif Aslam", "Arijit Singh")
print(text)

My favorite Singer is Arijit Singh.


### Strip

In [37]:
# strip() ---> it will rmove unnecssary spaces from string
name = "         Huzaifa          "
print(name.strip())

Huzaifa


In [38]:
name

'         Huzaifa          '

In [39]:
name = name.strip()
print(name)

Huzaifa


In [41]:
# lstrip() ---> remove annoying spaces from left of the string
name1 = "        Huzitech"
print(name1.lstrip())

Huzitech


In [42]:
# rstrip() ---> remove annoying spaces from right of the string
name1 = "Huzitech             "
print(name1.rstrip())

Huzitech
